dht.go

```diff
@@ -389,6 +390,15 @@ func (dht *IpfsDHT) Close() error {
 	return dht.proc.Close()
 }
 
+func (dht *IpfsDHT) protocolStrs() []string {
+	pstrs := make([]string, len(dht.protocols))
```

Comment:
You probably want `pstrs := make([]string, 0, len(dht.protocols))` (or want to assign by index instead of appending).